netket/operator/_discrete_operator.py

```diff
@@ -277,7 +278,18 @@ def _op__rmatmul__(self, other):
         return NotImplemented
 
     def to_linear_operator(self):
-        return self.to_sparse()
+        """
+        Convert the operator to a scipy.sparse.linalg.LinearOperator
+
+        Default implementation which uses the sparse matrix representation
+        internally.
+        """
+        A = self.to_sparse()
+        return scipy_LinearOperator(
+            A.shape,
+            A.__matmul__,
+            dtype=A.dtype,
+        )
```

Comment:
isn't this 99% to what we are already doing?

Reply:
if this was implemented as an on-the-fly matrix product it would make sense to keep it as it is and maybe leave the user the option to choose what mode to use ?

Reply:
it's just us doing it instead of scipy.

how about re raise a notImpmenentedError here, and move this conversion to lanczos_ed where I check if it's a jax operator?

Reply:
for the moment being (i.e. jax not supporting sparse eigensolvers for sparse matrices, if I understand correctly) I would just automatically convert the jax operator into a sparse scipy matrix. later I would instead rely on the jax sparse solver directly. if one day we decide to implement the matrix-free solver we can re-evaluate